modules/database/src/admin/schema.admin.ts

```diff
@@ -38,49 +43,75 @@ export class SchemaAdmin {
+    return {
+      schemas: cmsSchemas.concat(cndSchemas).map((schema: ConduitDatabaseSchema) => ({
+        ...schema,
+        extensions: schema.extensions.filter(ext => ext.ownerModule === 'database'),
+      })),
+    };
   }
 
-  async importCustomSchemas(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
+  async importSchemas(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
     const schemas = call.request.params.schemas;
     for (const schema of schemas) {
-      const existingSchema = await this.database
+      const found = await this.database
         .getSchemaModel('_DeclaredSchema')
         .model.findOne({ name: schema.name });
-      const operation = isNil(existingSchema) ? 'create' : 'update';
-      const imported = schema.modelOptions.conduit.imported === true;
-      const modelOptions = SchemaConverter.getModelOptions({
-        cmsSchema: true,
-        existingModelOptions: schema.modelOptions,
-        importedSchema: imported,
-      });
-      try {
-        validateSchemaInput(schema.name, schema.fields, modelOptions);
-      } catch (err: unknown) {
-        throw new GrpcError(status.INTERNAL, (err as Error).message);
-      }
-      await this.schemaController.createSchema(
-        new ConduitSchema(
-          schema.name,
-          schema.fields,
-          modelOptions,
-          schema.collectionName,
-        ),
-        operation,
-        imported,
-      );
-      if (schema.extensions.length > 0) {
-        for (const extension of schema.extensions) {
-          await this.database.setSchemaExtension(
-            schema.name,
-            extension.owner,
-            extension.fields,
-          );
+      const isCmsSchema = found
+        ? !!found.modelOptions.conduit.cms
+        : !!schema.modelOptions?.conduit?.cms;
+      if (isCmsSchema) {
+        const existingSchema = await this.database
+          .getSchemaModel('_DeclaredSchema')
+          .model.findOne({ name: schema.name });
+        const operation = isNil(existingSchema) ? 'create' : 'update';
+        const imported = schema.modelOptions.conduit.imported;
+        const modelOptions = SchemaConverter.getModelOptions({
+          cmsSchema: true,
+          existingModelOptions: schema.modelOptions,
+          importedSchema: imported,
+        });
+        try {
+          validateSchemaInput(schema.name, schema.fields, modelOptions);
+        } catch (err: unknown) {
+          throw new GrpcError(status.INTERNAL, (err as Error).message);
         }
+        await this.schemaController.createSchema(
+          new ConduitSchema(
+            schema.name,
+            schema.fields,
+            modelOptions,
+            schema.collectionName,
+          ),
+          operation,
+          imported,
+        );
       }
+      const extension = (schema.extensions as DeclaredSchemaExtension[]).find(
+        ext => ext.ownerModule === 'database',
+      );
+      await this.database.setSchemaExtension(
```

Comment:
This would crash for non-cms schemas that have yet to be created.
We should either:
a) check these beforehand and crash early
b) keep track of missing module schemas and alter the endpoint's response from `string` to `{ status: string, missing: string[] }` (or similar), where status would indicate success or partial success.

Let's discuss this before implementation.